pkg/virt-handler/vm.go

```diff
@@ -470,58 +467,34 @@ func (d *VirtualMachineController) hasTargetDetectedDomain(vmi *v1.VirtualMachin
 }
 
 func (d *VirtualMachineController) clearPodNetworkPhase1(vmi *v1.VirtualMachineInstance) {
-	// no need to cleanup with empty uid
 	if string(vmi.UID) == "" {
 		return
 	}
-	d.phase1NetworkSetupCache.Delete(vmi.UID)
-
-	// Clean Pod interface cache from map and files
-	d.podInterfaceCache.DeleteAllForVMI(vmi.UID)
-
-	err := d.networkCacheStoreFactory.CacheForVMI(vmi).Remove()
+	err := d.networkController.Teardown(vmi, func() error {
+		d.podInterfaceCache.DeleteAllForVMI(vmi.UID)
+		return d.networkCacheStoreFactory.CacheForVMI(vmi).Remove()
+	})
 	if err != nil {
 		log.Log.Reason(err).Errorf("failed to delete VMI Network cache files: %s", err.Error())
 	}
 }
 
-// Reaching into the network namespace of the VMI's pod is expensive because
-// it results in killing/spawning a posix thread. Only do this if it
-// is absolutely necessary. The cache informs us if this action has
-// already taken place or not for a VMI
```

Comment:
I think this comment is actually important - if only a bit unrelated.

Would you consider moving it along somewhere appropriate ?

Reply:
> I must admit that this comment makes me uncomfortable, as it is totally breaking the reconcile logic, specifically the actual state is no longer query but masked by a cache

And that's why it is important: it points out the oddities behind the cache. 

Sorry for having missed the comment you moved. All is fine.

Reply:
To me it points out why we should not do it (have the cache), but that is a different topic :)